### DISCUSSION TITLE
Support OAuth

### DISCUSSION THREAD

**Discussion by aaronpk** [bluesky-social/atproto#1154] [Protocol (atproto)]
There are many advantages to Bluesky/AT Proto supporting OAuth rather than having passwords be the only mechanism for getting access tokens for use at the API:

* Enable support for more secure authentication methods than passwords, such as multifactor authentication as well as passkeys
* Stop training users that it's okay to share their passwords with random applications
* etc etc

I wrote up some more detailed thoughts about how Bluesky can leverage OAuth in a blog post here:

https://aaronparecki.com/2023/03/09/5/bluesky-and-oauth

**Comment by mikestaub** [comment]
Another interesting area to watch, though still early, is Google's FedCM protocol: https://developers.google.com/privacy-sandbox/3pcd/fedcm

**Comment by DavidBuchanan314** [comment]
There's now a proposal for this: https://github.com/bluesky-social/proposals/tree/main/0004-oauth
